### DISCUSSION TITLE
Introduced Nightly Builds

### DISCUSSION THREAD

**Discussion by IHateMyKite** [IHateMyKite/UnforgivingDevices#129] [General]
As of now, Nightly builds should be implemented and working. Took only thousand of commits, but should be worth it. 
Every time there is push to main branch, the action will fire up. The action will build all papyrus scripts, and then upload the whole ziped project to release with [NB tag](https://github.com/IHateMyKite/UnforgivingDevices/releases/tag/NB). This means there will be only the latest version. Ziped file will always be named UnforgivingDevices_NB.zip.

This should make it easier for less experience users to try latest in-dev versions without needing to know how to build the script. The whole build takes only around 15 minutes.

The only possible issue is that the If there will be many pushes at once, there will be queue of many builds. It might be better to make it at it only build project once per day if there was push. Sadly, I don't know how to do it, so for now this should suffice.

I'm opened to any feedback and ideas.

**Comment by kurotatsu77** [comment]
This will surely help expand our experimental testers populace, great work!

About the schedule - maybe just make it everyday night build, no matter if there were any changes or not?

Any plans to make it compile the full FOMOD file for stand-alone install, to include all repositories (esp pathes, Unforgiving Skyrim, other modules etc)?

**Comment by IHateMyKite** [comment]
Making it once per 24 might be enough, so it at least don't make nb every time there is push. Regarding the FOMOD, I was also thinking about adding it, but I had so much working making this work that I give up on that idea for now. Hopefully it should be no hard. All it should take is to copy the files from fomod repo and then do some folder manipulations

**Comment by IHateMyKite** [comment]
Quick note. I have moved all required sources for compiling to new private repository, so I can avoid issues with licensing. Issue is that in case someone want to use NB on their fork, they needs to be collaborator on the said repository. Lastly, you also need to add new new secret for your fork with access token.
![pic](https://user-images.githubusercontent.com/103142567/224534086-8af1a73d-7c40-4a7a-bd1d-890e886ea80b.png)
It have to be named ACCESS_TOKEN and it should be filled with token with REPOSITORY scope. Said token can be generated here https://github.com/settings/tokens/new
Name is not important, only select correct scope 
![pic](https://user-images.githubusercontent.com/103142567/224534327-a8ff8bfd-2ac4-4516-82ba-c881e3bf1b60.png)
The generated value should then be copied to the repository secret.
For now, I added all existing contributors from UD to the private repository too. If some other person also want to have NB, please contact me.

**Comment by IHateMyKite** [comment]
Nwm, reworked it again, the papyrus repo is now public. Had to do some changes to sexlab scripts tho. Previous comment is now invalid
